sycl/test/filter_selector/select_device.cpp

```diff
@@ -0,0 +1,80 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RU: env SYCL_DEVICE_FILTER="*" %t.out
+// RU: env SYCL_DEVICE_FILTER=cpu %t.out
+// RU: env SYCL_DEVICE_FILTER=level_zero:gpu %t.out
+// RU: env SYCL_DEVICE_FILTER=opencl:gpu %t.out
+// RUN: env SYCL_DEVICE_FILTER=cpu,level_zero:gpu %t.out
+// RUN: env SYCL_DEVICE_FILTER=opencl:acc:0 %t.out
+//
+// Checks if only specified device types can be acquired from select_device
+// when SYCL_DEVICE_FILTER is set
+// Checks that no device is selected when no device of desired type is
+// available.
+//
+// REQUIRES: opencl,level_zero,host,cpu,gpu,accelerator
```

Comment:
the test will never execute because we select either level-zero or opencl plugin in LIT infra

Reply:
Yes, in current CI setting, it is not possible to run this.
Once we deprecate legacy env vars, the user should be able to use level_zero and opencl devices.